addons/website/models/website.py

```diff
@@ -1488,22 +1460,28 @@ def _check_snippet_used(self, snippet_occurences, asset_type, asset_version):
         return False
 
     def _disable_unused_snippets_assets(self):
-        snippets_assets = self._get_snippets_assets()
+        snippet_assets = self.env['ir.asset'].with_context(active_test=False).search_fetch(
+            [('path', 'like', '/static%/snippets/')],
+            ['active', 'path'], order='id')
+        snippet_re = re.compile(r'(\w*)\/.*\/snippets\/(\w*)\/(\d{3})(?:_\w*)?\.(js|scss)')
+        # regex will match /module/static/[.../]/snippets/snippet_id/XXX[_variable].asset_type
+        # _variable is not kept since only module, snippet_id, asset_version (XXX), asset_type are relevant
         html_fields_attributes = self._get_html_fields_attributes()
-
-        for snippet_module, snippet_id, asset_version, asset_type, _asset_id in snippets_assets:
-            is_snippet_used = self._is_snippet_used(snippet_module, snippet_id, asset_version, asset_type, html_fields_attributes)
-
-            # The regex catches XXX.scss, XXX.js and XXX_variables.scss
-            assets_regex = f'{snippet_id}/{asset_version}.+{asset_type}'
-
-            # The query will also set to active or inactive assets overrides, as they
-            # share the same snippet_id, asset_version and filename_type as their parents
-            self.env.cr.execute("""
-                UPDATE ir_asset
-                SET active = %(active)s
-                WHERE path ~ %(assets_regex)s
-            """, {"active": is_snippet_used, "assets_regex": assets_regex})
+        snippet_used = {}
+        for snippet_asset in snippet_assets:
+            match = snippet_re.match(snippet_asset.path)
+            if not match:
+                continue
+            (snippet_module, snippet_id, asset_version, asset_type) = match.groups()
+            if asset_type == 'scss':
+                asset_type = 'css'
+            key = (snippet_id, asset_version, asset_type)  # module is not relevant, we want the first one in the asset id order to filter module extension
+            if key not in snippet_used:
+                snippet_used[key] = self._is_snippet_used(snippet_module, snippet_id, asset_version, asset_type, html_fields_attributes)
+            is_snippet_used = snippet_used[key]
```

Comment:
I am not sure how useful (or necessary) was it back then to filter out snippet assets extensions/overrides.
I just found out this [part](https://github.com/odoo/odoo/pull/73005#discussion_r700143523) when checking the initial PR, which doesn't help much.

This case is quite verbose to handle now that this is written in python/orm instead of SQL, but let's keep the code as is, it's fine.
Snippet overrides is quite rare, so I guess it would make sense to do small amount of duplicate/useless work if it makes the code easier to read.

So: **ignore this comment**, I just wanted to write it for tracking purpose, should we work on this later again

```suggestion
            is_snippet_used = self._is_snippet_used(snippet_module, snippet_id, asset_version, asset_type, html_fields_attributes)
```
(and remove `snippet_used = {}` above)

Reply:
Even if we ignore impact on performance, a warning was logged 
`Cannot load template website_mass_mailing.s_popup: External ID not found in the system: website_mass_mailing.s_popup`

So this was needed in the current_state of _is_snippet_used